src/steps/do-query/do-get.js

```diff
@@ -63,20 +64,56 @@ export default function(requestContext, responseContext, registry) {
 
 }
 
-function parseFields(fieldsParam) {
+function parseFields(fieldsParam, registry) {
   let fields;
   if(typeof fieldsParam === "object") {
     fields = {};
     let isField = (it) => !arrayContains(["id", "type"], it);
 
     for(let type in fieldsParam) {
-      let provided = parseCommaSeparatedParam(fieldsParam[type]) || [];
+      let provided = parseCommaSeparatedParam(fieldsParam[type], type, registry) || [];
       fields[type] = provided.filter(isField);
     }
   }
   return fields;
 }
 
-function parseCommaSeparatedParam(it) {
-  return it ? it.split(",").map(decodeURIComponent) : undefined;
+function parseCommaSeparatedParam(it, type, registry) {
+  return it ? it.split(",").map(f => {
+    return transformDotSeparatedPath(f, type, registry);
+  }) : undefined;
+}
+
+function transformDotSeparatedPath(it, type, registry) {
+  let parts = it.split(".");
+  let types = [type];
+  parts.forEach((p, index) => {
+    if (index < parts.length - 1) {
+      types.push(registry.dbAdapter(types[index]).constructor.getReferencedType(types[index], p));
+    }
+  });
+
+  return parts.map((part, index) => {
+    if (shouldCamelizeType(types[index], registry)) {
+      return camelize(decodeURIComponent(part));
+    }
+    return decodeURIComponent(part);
+  }).join(".");
+}
+
+function transformFilters(it, type, registry) {
```

Comment:
I'm not sure if, in general, camelizing filters makes sense, as filter names may not correspond to field names. But, for now, this makes sense since we only have the simple filtering strategy, in which the keys are field names. Therefore, though, this code should definitely be taking the exceptions for the type into account.

Also, as a small stylistic note, I generally use `it` as a parameter name only when the function name is singular, so it's clear that the `it` corresponds to that thing in the function name. That is, if this function were called `transformFilter`, `it` would be the filter. When the function name is plural, I suppose I could use `them`, but I generally use a more descriptive name instead (like `fieldsParam` in the `parseFields` function above). So, I'd probably name this function `transformSimpleFilters` and then use `filters` as the argument name and `let filter in filters` in the loop.